tools/depmod.c

```diff
@@ -2310,32 +2304,37 @@ static int output_symbols_bin(struct depmod *depmod, FILE *out)
 
 	hash_iter_init(depmod->symbols, &iter);
 
+	strbuf_init_with_stack(&salias, 1024);
+	strbuf_pushmem(&salias, base, baselen);
+
 	while (hash_iter_next(&iter, NULL, &v)) {
 		int duplicate;
 		const struct symbol *sym = v;
-		size_t len;
-		char *s;
+		size_t sym_sz;
 
 		if (sym->owner == NULL)
 			continue;
 
-		len = strlen(sym->name);
+		sym_sz = strlen(sym->name) + 1;
 
-		if (scratchbuf_alloc(&salias, baselen + len + 1) < 0) {
+		if (!strbuf_reserve(&salias, sym_sz)) {
 			ret = -ENOMEM;
-			goto err_scratchbuf;
+			goto err_alloc;
 		}
-		s = scratchbuf_str(&salias);
-		memcpy(s + baselen, sym->name, len + 1);
-		duplicate = index_insert(idx, s, sym->owner->modname, sym->owner->idx);
+
+		strbuf_pushmem(&salias, sym->name, sym_sz);
```

Comment:
`sym_sz` is `strlen(sym->name) + 1` so here we already add the ending NUL byte. If we skip it, even this one could become `strbuf_pushchars`.

Reply:
I was adding the '\0' on purpose to match the previous code, but indeed we can keep that out.